assembly/src/library/mod.rs

```diff
@@ -354,7 +352,7 @@ mod use_std_library {
             }
 
             if modules.is_empty() {
-                return Err(LibraryError::Empty(namespace.clone()).into());
+                return Err(LibraryError::EmptyModules(namespace.clone()).into());
```

Comment:
Same deal here IMO - not sure we need to raise an error for this, it is basically harmless if the library is empty, and in practice would be caught when an attempt is made to introspect/use the library, where the error would have more meaning (i.e. you can't do X because the library you are using has no module/procedure/whatever that you need).

In other words, we're sort of ascribing more meaning than necessary to what a `Library` _is_, where I'd rather apply those sorts of validation rules at a higher level, i.e some kind `Package` type. Making those validation rules a part of what constitutes a valid package (and would include things like names, dependency info, etc.).

Reply:
Removed this error and added a `TODO` comment to emit warnings in the future when we try to output an empty library.